### ISSUE TITLE
[EPIC] Support OAuth User Login via the UI in OSS

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#13021]
## Tell us about the problem you're trying to solve
Users want to be able to do oauth in the UI in Airbyte OSS. This ticket tracks that effort. 

## Describe the alternative you’ve considered or used
Current alternative is to do oauth completey outside airbyte like described in airbytehq/airbyte#768

**Comment by andyjih** [comment]
@evantahler Removed the `team/workflows` label from this one. Let me know if that's right!

**Comment by winterrobert** [comment]
What are the current blockers for getting this into the OSS version - would you be open to accepting community support to implement it or is it to closely tied to Airbyte Cloud?

**Comment by ivicaned** [comment]
I agree that this would be very nice and useful feature for Open Source edition too, as most external system requires OAuth flows, and also many companies force it for all integrations for security reasons.

**Comment by andyjih** [comment]
@winterrobert @ivicaned Would either of you be open to chatting with me on a call to discuss this feature/topic a bit more?

**Comment by jx2lee** [comment]
[DataHub](https://datahubproject.io/docs/authentication/guides/sso/configure-oidc-react/), a data discovery platform, also offers OIDC capabilities. I also agree that it is a necessary feature to evolve with the community.

**Comment by alexkreidler** [comment]
[Nango](https://github.com/NangoHQ/nango) is a new open-source project that has figured out the quirks[^1] in [over 100 OAuth flows](https://github.com/NangoHQ/nango/blob/master/packages/shared/providers.yaml) and has a simple SDK/API for [creating a Connection (doing the OAuth flow) on the frontend](https://docs.nango.dev/sdks/frontend) and [using the connection (accessing the tokens to call the API)](https://docs.nango.dev/sdks/node).

I think we should leverage that work so we don't have to duplicate it in Airbyte. If we allow users to pass in OAuth tokens, then they could generate them themselves, or via their code, or with a tool like Nango. We could still have modules that handle OAuth flows for sites not supported by Nango.

[^1]: They wrote [a great blog post](https://www.nango.dev/blog/why-is-oauth-still-hard) called "Why is OAuth still hard in 2023?" with examples of the quirks in the Jira, Notion, Shopify and Zendesk flows.

**Comment by MrPatrickHenry** [comment]
Hi where did this land, kinda feel its major need on OSS platform

**Comment by jorgeparavicini** [comment]
Is there an update on this?

**Comment by marcosmarxm** [comment]
Airbyte open source supports now a single-user basic auth mechanism. Any other auth like Single Sign-On (SSO) is an enterprise feature.

**Comment by reubano** [comment]
When you say "basic auth", does that include OAuth?
